contracts/poolMaster/PoolMaster.sol

```diff
@@ -0,0 +1,329 @@
+   */
+  function liquidateTokensToKnc(IERC20Ext[] calldata tokens, uint256[] calldata minRates)
+    external
+    onlyOperator
+  {
+    require(tokens.length == minRates.length, 'unequal lengths');
+    for (uint256 i = 0; i < tokens.length; i++) {
+      if (tokens[i] == ETH_ADDRESS) {
+        kyberProxy.swapEtherToToken{value: address(this).balance}(newKnc, minRates[i]);
+      } else if (tokens[i] != newKnc) {
+        // token allowance should have been given
+        kyberProxy.swapTokenToToken(
+          tokens[i],
+          tokens[i].balanceOf(address(this)),
+          newKnc,
+          minRates[i]
+        );
+      }
+    }
+    uint256 availableKnc = _administerAdminFee(FeeTypes.CLAIM, getAvailableNewKncBalanceTwei());
+    _stake(availableKnc);
+  }
+
+  /*
+   * @notice Called by admin on deployment for KNC
+   * @dev Approves Kyber Proxy contract to trade KNC
+   * @param Token to approve on proxy contract
+   * @param Pass _giveAllowance as true to give max allowance, otherwise resets to zero
+   */
+  function approveKyberProxyContract(IERC20Ext token, bool giveAllowance) external onlyOperator {
+    require(token != newKnc, 'knc not allowed');
+    uint256 amount = giveAllowance ? type(uint256).max : 0;
+    token.safeApprove(address(kyberProxy), amount);
+  }
+
+  function withdrawAdminFee() external onlyOperator {
+    uint256 fee = withdrawableAdminFees.sub(1);
+    withdrawableAdminFees = 1;
+    newKnc.safeTransfer(admin, fee);
+  }
+
+  function stakeAdminFee() external onlyOperator {
+    uint256 fee = withdrawableAdminFees.sub(1);
+    withdrawableAdminFees = 1;
+    _deposit(fee, admin);
+  }
+
+  /*
+   * @notice Returns KNC balance staked to the DAO
+   */
+  function getLatestStake() public view returns (uint256 latestStake) {
+    (latestStake, , ) = kyberStaking.getLatestStakerData(address(this));
+  }
+
+  /*
+   * @notice Returns KNC balance available to stake
+   */
+  function getAvailableNewKncBalanceTwei() public view returns (uint256) {
+    return newKnc.balanceOf(address(this)).sub(withdrawableAdminFees);
+  }
+
+  /*
+   * @notice Returns fee (in basis points) depending on fee type
+   */
+  function getFeeRate(FeeTypes _type) public view returns (uint256) {
+    if (_type == FeeTypes.MINT) return adminFees.mintFeeBps;
+    else if (_type == FeeTypes.CLAIM) return adminFees.claimFeeBps;
+    return adminFees.burnFeeBps;
+  }
+
+  /*
+   * @notice For APY calculation, returns rate of 1 pool master token to KNC
+   */
+  function getProRataKnc() public view returns (uint256) {
+    if (totalSupply() == 0) return 0;
+    return getLatestStake().mul(PRECISION).div(totalSupply());
+  }
+
+  function _changeFees(
+    uint256 _mintFeeBps,
+    uint256 _claimFeeBps,
+    uint256 _burnFeeBps
+  ) internal {
+    require(_mintFeeBps <= BPS, 'bad mint bps');
+    require(_claimFeeBps <= BPS, 'bad claim bps');
+    require(_burnFeeBps >= 10 && _burnFeeBps <= BPS, 'bad burn bps');
```

Comment:
Pls set another upper bounds for _burnFeeBps. For example, pool master can set _burnFeeBps up tp max to block withdrawing

Reply:
Pls create an event for this event too

Reply:
what's a good upper bound? 10%?

Reply:
1 or 2%

Reply:
I think all fees should be at most 10%

Reply:
i'll go with a sanity fee rate of 10% for all fees